secboot/bootstrap_container_sb.go

```diff
@@ -66,6 +67,21 @@ func (bc *bootstrappedContainer) AddKey(slotName string, newKey []byte) error {
 	return nil
 }
 
+func (bc *bootstrappedContainer) AddRecoveryKey(slotName string, rkey keys.RecoveryKey) error {
+	if bc.finished {
+		return fmt.Errorf("internal error: bootstrapped container was a already finished")
+	}
+
+	if slotName == "" {
+		slotName = "default-recovery"
+	}
+
+	if err := sbAddLUKS2ContainerRecoveryKey(bc.devicePath, slotName, sb.DiskUnlockKey(bc.key), sb.RecoveryKey(rkey)); err != nil {
```

Comment:
I see we use `keys.RecoveryKey` and then we convert to `sb.RecoveryKey`, just to add the key. But I do not see where we call `sb.RecoveryKey.String()`. So I am wondering who is doing the translation to the string from bytes. Are we just sending bytes to the client and they have to do the conversion?

Reply:
Nevermind, it is in `secboot/keys/keys_sb.go`.